### PR TITLE
feat(scheduler): Create V2 IntervalAction DTO and Model

### LINKED ISSUE #517: [V2 schedular] Create IntervalAction DTO and Model
Create IntervalAction DTO and Model for V2 API

### PR DESCRIPTION
**Pull request by weichou1229**
Close #517

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Create IntervalAction DTO and Model for scheduler V2 API

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

File: v2/dtos/intervalaction.go [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+
+// IntervalAction and its properties are defined in the APIv2 specification:
+// https://app.swaggerhub.com/apis-docs/EdgeXFoundry1/support-scheduler/2.x#/IntervalAction
+type IntervalAction struct {
+	common.Versionable `json:",inline"`
+	Created            int64  `json:"created,omitempty"`
+	Modified           int64  `json:"modified,omitempty"`
+	Id                 string `json:"id,omitempty" validate:"omitempty,uuid"`
+	Name               string `json:"name" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
+	IntervalName       string `json:"intervalName" validate:"edgex-dto-none-empty-string,edgex-dto-rfc3986-unreserved-chars"`
```
Comment (lenny-intel) [inline v2/dtos/intervalaction.go:21]: Why both Name and IntervalName?

File: v2/dtos/intervalaction.go [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+	Protocol           string `json:"protocol,omitempty"`
+	Host               string `json:"host,omitempty"`
+	Port               int    `json:"port,omitempty"`
+	Path               string `json:"path,omitempty"`
+	Parameters         string `json:"parameters,omitempty"`
+	HTTPMethod         string `json:"httpMethod,omitempty" validate:"omitempty,oneof='GET' 'HEAD' 'POST' 'PUT' 'DELETE' 'TRACE' 'CONNECT'"`
+	User               string `json:"user,omitempty"`
+	Password           string `json:"password,omitempty"`
```
Comment (lenny-intel) [inline v2/dtos/intervalaction.go:29]: @cloudxxx8 , @jpwhitemn , @hutchic , Looks like have need for Scheduler to store these secrets in Vault.

File: v2/models/intervalaction.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+	Name         string
+	IntervalName string
+	Protocol     string
+	Host         string
+	Port         int
+	Path         string
+	Parameters   string
+	HTTPMethod   string
+	User         string
+	Password     string
```
Comment (lenny-intel) [inline v2/models/intervalaction.go:23]: Note that in Secure mode the password should not be stored in DB, but instead stored in SecretStore. 
@hutchic,  We need to add this to the Security WG agenda

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: v2/dtos/intervalaction.go (hunk 1 above)
Comment (weichou1229) [reply v2/dtos/intervalaction.go:21]: It's interval's name which refer to the associated interval.

Comment (lenny-intel) [reply v2/dtos/intervalaction.go:21]: Right! Thansk!

**Review by lenny-intel — APPROVED** [review]
LGTM

File: v2/dtos/intervalaction.go (hunk 2 above)
Comment (cloudxxx8) [reply v2/dtos/intervalaction.go:29]: @lenny-intel we need re-design the interval-action.  V1 doesn't support MQTT, but there are lots of message bus fields here.  I prefer to support pure REST without security in the initial version.

Comment (lenny-intel) [reply v2/dtos/intervalaction.go:29]: @cloudxxx8 , thanks for the clarification. I agree with REST only until we have a use case that requires MQTT.

File: v2/models/intervalaction.go (hunk 3 above)
Comment (lenny-intel) [reply v2/models/intervalaction.go:23]: @hutchic , no longer needed as agenda item after @cloudxxx8's clarification.

**Pull request merged**